src/Imdb/Calendar.php

```diff
@@ -0,0 +1,92 @@
+# IMDBPHP UPCOMMING RELEASES (based on IMDBPHP TOP CHARTS (c)Ricardo Silva) #
+# written/modified/extended by Ed (github user: duck7000)                   #
+# ------------------------------------------------------------------------- #
+# This program is free software; you can redistribute and/or modify it      #
+# under the terms of the GNU General Public License (see doc/LICENSE)       #
+#############################################################################
+
+namespace Imdb;
+
+/**
+ * Obtaining information about upcomming movie releases as seen on imdB
+ * https://www.imdb.com/calendar
+ * @author Ricardo Silva (banzap) <banzap@gmail.com>
+ * @author Ed (github user: duck7000)
+ */
+class Calendar extends MdbBase
+{
+
+    protected $page = null;
+
+    /**
+     * Get upcomming movie releases as seen on imdB
+     * @return array (array[string date, array releases] of arrays[title,year,imdbid])
+     * @parameter $country, This defines wich country list is returned
+     * for example de, nl, us as they appear on https://www.imdb.com/calendar
+     */
+    public function upcommingReleases($country)
+    {
+        $page = $this->getXpathPage($country);
+        $calendar = array();
+        $dates = $page->query("//*[@id='main']/h4");
+        foreach ($dates as $key => $date) {
+            $key++;
+            $titlesRaw = $page->query("//*[@id='main']/ul[$key]//li");
+            $titles = array();
+            foreach ($titlesRaw as $value) {
+                $href = $value->getElementsByTagName('a')->item(0)->getAttribute('href');
+                preg_match('!.*?/title/tt(\d+)/.*!', $href, $imdbid);
+                $title = trim($value->getElementsByTagName('a')->item(0)->nodeValue);
+                preg_match('#\((.*?)\)#', $value->nodeValue, $year);
+                $titles[] = array(
+                    'title' => $title,
+                    'year' => $year[1],
+                    'imdbid' => $imdbid[1]
+                );
+            }
+            $calendar[] = array(
+                'date' => trim($date->nodeValue),
+                'releases' => $titles
+            );
+        }
+        var_dump($calendar[0]);
+        return $calendar;
+    }
+
+    protected function buildUrl($context = null)
+    {
+        return "https://" . $this->config->imdbsite . "/calendar/?region=$context";
+    }
+    
+    /**
+     * @param string $page
+     * @return \DomXPath
+     */
+    protected function getXpathPage($page)
+    {
+        if (!empty($this->xpathPage[$page])) {
+            return $this->xpathPage[$page];
+        }
+        $source = $this->getPage($page);
+        libxml_use_internal_errors(true);
+        /* Createa a new DomDocument object */
+        $dom = new \DomDocument;
+        /* Load the HTML */
+        $dom->loadHTML($source);
+        /* Create a new XPath object */
+        $this->xpathPage[$page] = new \DomXPath($dom);
+        return $this->xpathPage[$page];
+    }
```

Comment:
You can remove this, I've moved it into MdbBase so there's no need for it to be duplicated
```suggestion
```

Reply:
I based this on Charts.php and assumed that if i want to use xpath this method was needed. So fine i'll delete it